### PR TITLE
tests: fixtures refactoring

### LINKED ISSUE #40: tests: refactor tests to make proper use of fixtures

### PR DESCRIPTION
**Pull request by JavierDelgadoFernandez**
* Refactorizes the tests in order to use the conftest fixtures.
  (closes #40)

Signed-off-by: Javier Delgado <javier.delgado.fernandez@cern.ch>

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -106,3 +108,21 @@ def teardown():
 
     request.addfinalizer(teardown)
     return app
+
+
+@pytest.fixture()
+def communities_creation(app):
+    """Create some example communities."""
+    with app.app_context():
```
Comment (lnielsen) [inline tests/conftest.py]: It's better to refactor the app fixture to use the pattern here: https://github.com/inveniosoftware/invenio-records-rest/blob/master/tests/conftest.py#L172-L173 Also the DB fixture will be useful: https://github.com/inveniosoftware/invenio-records-rest/blob/master/tests/conftest.py#L179-L190

File: tests/conftest.py [hunk 2]
```diff
@@ -106,3 +108,21 @@ def teardown():
...
+        user1 = create_test_user()
```
Comment (lnielsen) [inline tests/conftest.py]: The fixture can be split into 1) test user creation and 2) community creation (which can depend on test user(s))

File: tests/conftest.py [hunk 3]
```diff
@@ -106,3 +108,21 @@ def teardown():
 
     request.addfinalizer(teardown)
     return app
+
+
+@pytest.fixture()
+def communities_creation(app):
```
Comment (lnielsen) [inline tests/conftest.py]: ``communities_creation`` -> ``communities``

File: tests/test_invenio_communities.py [hunk 4]
```diff
@@ -199,56 +199,25 @@ def test_oaipmh_sets(app):
...
-                                     'communities/comm2/'),
-                        },
-                    },
-                ],
-                'total': 2,
-            }
-            assert len(['links']) == 1
+    with app.test_client() as client:
+        response = client.get('/api/communities/')
+        response_data = json.loads(response.get_data(as_text=True))
```
Comment (lnielsen) [inline tests/test_invenio_communities.py]: this line appears many places...

```
def get_json(res, code=None):
     if code:
         assert res.status_code == code
     return json.loads(res.get_data(as_text=True))
```

File: tests/test_invenio_communities.py [hunk 5]
```diff
@@ -199,56 +199,25 @@ def test_oaipmh_sets(app):
...
+        assert response_data['hits']['total'] == 3
+        assert len(response_data['hits']['hits']) == 3
+
+        comm = response_data['hits']['hits'][0]
+        assert comm['description'] == 'Description1'
```
Comment (lnielsen) [inline tests/test_invenio_communities.py]: you could build a function ``assert_community_serialization()`` which takes care of this to simplify the test.

File: tests/test_invenio_communities.py [hunk 6]
```diff
@@ -274,65 +243,19 @@ def test_community_delete(app):
...
-        db.session.add(comm1)
-        db.session.add(comm2)
-        db.session.commit()
+    with app.test_client() as client:
+        response = client.get('/api/communities/?q=comm&sort=title')
+        response_data = json.loads(response.get_data(as_text=True))
+
+        assert response_data['hits']['total'] == 2
+        assert response_data['hits']['hits'][0]['id'] == 'comm2'
+        assert response_data['hits']['hits'][1]['id'] == 'comm1'
```
Comment (lnielsen) [inline tests/test_invenio_communities.py]: 👍

File: tests/test_invenio_communities.py [hunk 7]
```diff
@@ -366,57 +289,60 @@ def parse_path(app, path):
...
-            data = json.loads(response.get_data(as_text=True))
-            assert len(data['hits']['hits']) == 1
-            assert 'prev' in data['links']
-            assert 'next' not in data['links']
-
-
-def test_communities_rest_get_details(app):
+    with app.test_client() as client:
+        response = client.get('/api/communities/?size=1')
+        assert any_of(response.headers, lambda h: check_array_in_string(
```
Comment (lnielsen) [inline tests/test_invenio_communities.py]: This test method has a lot of boilerplate which can be further reduced

Comment (JavierDelgadoFernandez) [inline tests/test_invenio_communities.py]: @lnielsen Updated. What do you mean here?

**Comment by JavierDelgadoFernandez** [comment]
@lnielsen Updated.

**Pull request merged**